### PR TITLE
refactor(language-service): Replace TypeDiagnostic with ng.Diagnostic

### PR DESCRIPTION
**Pull request by kyliau**
This commit cleans up `expression_type.ts` by

1. Removing the unnecessary `TypeDiagnostic` class. It's replaced by
`ng.Diagnostic`.
2. Consolidating `reportError()` and `reportWarning()` to
`reportDiagnostic()`.

This is prep work so that we could make some of the type diagnostics a
suggestion in later PRs.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/language-service/src/expression_diagnostics.ts [hunk 1]
```diff
@@ -334,15 +326,14 @@ class ExpressionDiagnosticsVisitor extends RecursiveTemplateAstVisitor {
...
-  private diagnoseExpression(ast: AST, offset: number, includeEvent: boolean) {
-    const scope = this.getExpressionScope(this.path, includeEvent);
-    this.diagnostics.push(...getExpressionDiagnostics(scope, ast, this.info.query, {
-                            event: includeEvent
-                          }).map(d => ({
-                                   span: offsetSpan(d.ast.span, offset + this.info.offset),
-                                   kind: d.kind,
-                                   message: d.message
-                                 })));
+  private diagnoseExpression(ast: AST, offset: number, event: boolean) {
```
Comment (ayazhafiz) [inline packages/language-service/src/expression_diagnostics.ts]: nice!

File: packages/language-service/src/expression_type.ts [hunk 2]
```diff
@@ -399,23 +401,15 @@ export class AstType implements AstVisitor {
...
-      this.diagnostics.push(new TypeDiagnostic(ts.DiagnosticCategory.Error, message, ast));
-    }
-    return this.anyType;
-  }
-
-  private reportWarning(message: string, ast: AST): Symbol {
-    if (this.diagnostics) {
-      this.diagnostics.push(new TypeDiagnostic(ts.DiagnosticCategory.Warning, message, ast));
-    }
+  private reportDiagnostic(message: string, ast: AST, kind = ts.DiagnosticCategory.Error): Symbol {
```
Comment (ayazhafiz) [inline packages/language-service/src/expression_type.ts]: Can we get rid of the return here and explicitly return `this.anyType` where it needs to be? I think that would be much clearer.

Comment (kyliau) [reply packages/language-service/src/expression_type.ts]: done

**Review by ayazhafiz — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]
lgtm to make pullapprove happy

**Pull request closed without merging**